File: Data Structures & Algorithms/counting-bits/submission-1.py
class Solution:
    def countBits(self, n: int) -> List[int]:

        def count(n):
            res = 0
            while n:
                res+= n % 2 
                n = n // 2
            return res

        res = []
        for i in range(n + 1):
            res.append(count(i))
        return res
        

        

File: Data Structures & Algorithms/daily-temperatures/submission-1.java


class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Stack<int[]> stack = new Stack<>();
        int[] res = new int[temperatures.length];

        for (int i = 0; i < temperatures.length; i++) {
            int currentTemp = temperatures[i];

            while (!stack.isEmpty() && currentTemp > stack.peek()[1]) {
                int index = stack.pop()[0];
                res[index] = i - index;
            }

            stack.push(new int[]{i, currentTemp});
        }

        return res;
    }
}
